rafthttp/stream.go

```diff
@@ -374,11 +372,13 @@ func (cr *streamReader) dial(t streamType) (io.ReadCloser, error) {
 	req.Header.Set("X-Etcd-Cluster-ID", cr.cid.String())
 	req.Header.Set("X-Raft-To", cr.remote.String())
 
-	var peerURLs []string
-	for _, url := range cr.localPeer.urls() {
-		peerURLs = append(peerURLs, url.String())
+	if cr.tr.URLs != nil {
+		var peerURLs []string
+		for _, url := range cr.tr.URLs {
+			peerURLs = append(peerURLs, url.String())
+		}
+		req.Header.Set("X-Server-Peers", strings.Join(peerURLs, ","))
```

Comment:
Ideally we should add this logic to other types of senders too, like pipeline and snapshot sender. (It works for just adding to stream since the peer will always try to connect as stream anyway)